hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -1826,6 +1828,20 @@ public void updatePeerList(List<String> newPeers) {
         }
       }
     }
+    String leaderId = "";
+    if (isRatisEnabled) {
+      RaftPeer leader = null;
+      try {
+        leader = omRatisServer.getLeader();
+      } catch (IOException ex) {
+        LOG.error("IOException while getting the " +
+            "Ratis server leader.", ex);
+      }
+      if (Objects.nonNull(leader)) {
+        leaderId = leader.getId().toString();
+      }
+    }
+    omHAMetricsInit(leaderId);
```

Comment:
If `leaderId` is not set due to ratis not enabled or error getting leader a blank "" string is passed to `omHAMetricsInit`.  In `omHAMetrics` iin this case what happens?  We should handle that. (see also comment in `omHAMetrics.getMetrics()`)

Reply:
> If `leaderId` is not set due to ratis 

In that case `updatePeerList()` will never get called and `OMHAMetrics` won't be registered.

> error getting leader a blank "" string is passed to `omHAMetricsInit`

`OMHAMetrics` check whether the current `OMNodeId` is equal to `leaderId`. If `leaderId` is empty then this check will return false and state will be set to 0. If `leaderId` is empty, then there is no leader and setting state to follower seems reasonable to me.

Reply:
> If leaderId is empty, then there is no leader and setting state to follower seems reasonable to me.

Thanks @xBis7 for the comments on this.  In this case, the leader is undefined.  Undefined in neither follower nor leader.  We should indicate this as such in the `OMHAMetrics `- handling this empty string `leaderId` in the `OMHAMetrics.create()`, checking the condition here and _not_ callling `omHAMetricsInit()` or similar.

Reply:
> Undefined in neither follower nor leader.

OM Ratis Roles on the OM endpoint sets the current node's status to follower in that case. 

> checking the condition here and not callling omHAMetricsInit() or similar.

This seems best. I'll add a check.